maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/SpaInstaller.java

```diff
@@ -101,18 +105,21 @@ public void installFromDistroProperties(File appDataDir, DistroProperties distro
 
 		// First pull any optional properties that may be used to specify the core, node, or npm versions
 		// These properties are not passed to the build tool, but are used to specify the build execution itself
-		String coreVersion = spaBuildProperties.remove("core");
-		if (coreVersion == null) {
-			coreVersion = "next";
-		}
-		String nodeVersion = spaBuildProperties.remove("node");
-		if (nodeVersion == null) {
-			nodeVersion = NODE_VERSION;
-		}
-		String npmVersion = spaBuildProperties.remove("npm");
-		if (npmVersion == null) {
-			npmVersion = NPM_VERSION;
-		}
+		// Use defaults but allow overrides from distro.properties, system properties, or environment variables
+		String coreVersion = Optional.ofNullable(spaBuildProperties.remove("core"))
+				.orElse("next");
+
+		String nodeVersion = Optional.ofNullable(spaBuildProperties.remove("node.version"))
+				.orElseGet(() -> System.getProperty("sdk.node.version",
```

Comment:
Are we using both `sdk.node.version` and `spa.node.version`?

Reply:
oversight, these should be the same but can be used differently e.g something like `mvn clean install -Dspa.node.version=18.17.1` or setting it in distro.properties `spa.node.version=18.17.1` or export SPA_NODE_VERSION=18.17.1. 

Let me update that!!
Hope that answers you?